src/main/java/com/gooddata/md/visualization/VisualizationConverter.java

```diff
@@ -77,11 +94,30 @@ public static Afm convertToAfm(final VisualizationObject visualizationObject) {
      *
      * @param visualizationObject which will be converted to {@link Execution}
      * @param visualizationClassGetter {@link Function} for fetching VisualizationClass, which is necessary for correct generation of {@link ResultSpec}
+     * @throws Exception when VisualizationClass does not belong to provided VisualizationObject
      * @return {@link Execution} object
      */
     public static ResultSpec convertToResultSpec(final VisualizationObject visualizationObject,
-                                          final Function<String, VisualizationClass> visualizationClassGetter) {
-        VisualizationClass visualizationClass = notNull(visualizationClassGetter).apply(notNull(visualizationObject).getVisualizationClassUri());
+            final Function<String, VisualizationClass> visualizationClassGetter) throws Exception {
+        return convertToResultSpec(visualizationObject,
+                notNull(visualizationClassGetter).apply(notNull(visualizationObject).getVisualizationClassUri()));
+    }
+
+    /**
+     * Generate ResultSpec from Visualization object. Currently {@link ResultSpec}'s {@link Dimension}s can be generated
+     * for table and four types of chart: bar, column, line and pie.
+     *
+     * @param visualizationObject which will be converted to {@link Execution}
+     * @param visualizationClass VisualizationClass, which is necessary for correct generation of {@link ResultSpec}
+     * @throws Exception when VisualizationClass does not belong to provided VisualizationObject
+     * @return {@link Execution} object
+     */
+    public static ResultSpec convertToResultSpec(final VisualizationObject visualizationObject,
+            final VisualizationClass visualizationClass) throws Exception {
+        if (!visualizationObject.getVisualizationClassUri().equals(visualizationClass.getUri())) {
+            throw new Exception("VisualizationClass URI does not match the VisualizationObject URI, "
```

Comment:
Wouldn't it be nicer to have some specific exception type instead?

Reply:
It would be for sure, wanted to discuss this!